modules/rank.js

```diff
@@ -0,0 +1,164 @@
+            "ain't interested in no XP gettin'.",
+            "don't talk to me no more, so I ignore 'em."
+          ];
+          await interaction.editReply(`**${member}** ${u.rand(snark)}\n(Try \`/rank track\` if you want to participate in chat ranks!)`);
+        }
+      } else if (subcommand === "leaderboard") {
+        // View leaderboard
+        await interaction.deferReply();
+
+        const members = interaction.guild.members.cache;
+
+        const leaderboard = await Module.db.user.getLeaderboard({
+          members,
+          member: interaction.member
+        });
+
+        const records = leaderboard.map(l => `${l.rank}: ${members.get(l.discordId).toString()} (${l.currentXP.toLocaleString()} XP)`);
+        const embed = u.embed()
+        .setTitle("LDSG Season Chat Leaderboard")
+        .setThumbnail(interaction.guild.iconURL({ format: "png" }))
+        .setURL("https://my.ldsgamers.com/leaderboard")
+        .setDescription("Current season chat rankings:\n" + records.join("\n"))
+        .setFooter("Use `/rank track` to join the leaderboard!");
+
+        await interaction.editReply({ embeds: [embed] });
+      } else if (subcommand === "track") {
+        // Set XP tracking
+        await interaction.deferReply({ ephemeral: true });
+        const track = interaction.options.getBoolean("choice");
+        await Module.db.user.trackXP(interaction.user, track);
+        await interaction.editReply({
+          content: `Ok! I'll ${track ? "start" : "stop"} tracking your XP!`,
+          ephemeral: true
+        });
+      } else {
+        interaction.reply({
+          content: "Well, this is embarrasing. I don't know what you asked for.",
+          ephemeral: true
+        });
+      }
+    } catch (error) {
+      u.errorHandler(error, interaction);
+    }
+  }
+})
+.setInit((talking) => {
+  if (talking) {
+    for (const user of talking) active.add(user);
+  }
+})
+.setUnload(() => active)
+.addEvent("message", (msg) => {
+  if (
+    msg.guild?.id == Module.config.ldsg &&
+    msg.author &&
+    !active.has(msg.author.id) &&
+    !(Rank.excludeChannels.includes(msg.channel.id) || Rank.excludeChannels.includes(msg.channel.parentId)) &&
+    !msg.webhookId &&
+    !u.parse(msg) &&
+    !msg.author.bot
+  ) {
+    active.add(msg.author.id);
+  }
+})
+.setClockwork(() => {
+  // Do stuff
+  try {
+    return setInterval(async function(client) {
+      try {
+        const response = await Module.db.user.addXp(active);
+        if (response.users.length > 0) {
+          const ldsg = client.guilds.cache.get(Module.config.ldsg);
+          for (const user of response.users) {
+            const member = ldsg.members.cache.get(user.discordId);
+
+            // Remind mods to trust people!
+            if ((user.posts % 25 == 0) && !member.roles.cache.has(Module.config.roles.trusted) && !(member.roles.cache.has(Module.config.roles.trusted) || member.roles.cache.has(Module.config.roles.untrusted))) {
```

Comment:
Part of this statement is redundant.
```suggestion
            if ((user.posts % 25 == 0) && !member.roles.cache.has(Module.config.roles.trusted) && !member.roles.cache.has(Module.config.roles.untrusted)) {
```

Reply:
Looks like I started doing my logic table halfway through typing and forgot I had begun!